### PR TITLE
Ensure that native-sources package results in the configuration as native

### PR DESCRIPTION
**Pull request by geoand**
Furthermore, fail the build when native-sources is requested but the extension
creates its own output

Related to: https://github.com/quarkusio/quarkus/pull/15233#issuecomment-784849473

### CODE REVIEW

File: extensions/google-cloud-functions/deployment/src/main/java/io/quarkus/gcp/functions/deployment/CloudFunctionDeploymentBuildStep.java [hunk 1]
```diff
@@ -21,6 +23,12 @@ public UberJarRequiredBuildItem forceUberJar() {
         return new UberJarRequiredBuildItem();
     }
 
+    @BuildStep(onlyIf = NativeSourcesBuild.class)
+    void failForNativeSources(BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
+        throw new IllegalArgumentException(
+                "The Goog Cloud extensions are incompatible with the 'native-sources' package type.");
```
Comment (jerboaa) [inline extensions/google-cloud-functions/deployment/src/main/java/io/quarkus/gcp/functions/deployment/CloudFunctionDeploymentBuildStep.java:29]: Typo `Google`

Comment (geoand) [reply extensions/google-cloud-functions/deployment/src/main/java/io/quarkus/gcp/functions/deployment/CloudFunctionDeploymentBuildStep.java:29]: The surprising thing would be for a PR of mine to **not** contain a typo 😆

Comment (geoand) [reply extensions/google-cloud-functions/deployment/src/main/java/io/quarkus/gcp/functions/deployment/CloudFunctionDeploymentBuildStep.java:29]: Fixed

**Review by stuartwdouglas — APPROVED** [review]
I think this is ok for now, but this is really showing off the limitations of quarkus.package.type. 

We need something more flexible that allows for compositibility between the different outputs (e.g. native-source is a precursor to native so native should also imply native-source).

This needs a discussion on the dev list though.

**Comment by geoand** [comment]
> I think this is ok for now, but this is really showing off the limitations of quarkus.package.type.
> 
> We need something more flexible that allows for compositibility between the different outputs (e.g. native-source is a precursor to native so native should also imply native-source).
> 

Do you have something in mind?

**Pull request merged**